module/zfs/zio.c

```diff
@@ -3445,52 +3446,85 @@ zio_ddt_collision(zio_t *zio, ddt_t *ddt, ddt_entry_t *dde)
 }
 
 static void
-zio_ddt_child_write_ready(zio_t *zio)
+zio_ddt_child_write_done(zio_t *zio)
 {
 	ddt_t *ddt = ddt_select(zio->io_spa, zio->io_bp);
 	ddt_entry_t *dde = zio->io_private;
-	zio_t *pio;
 
-	if (zio->io_error)
-		return;
+	zio_link_t *zl = NULL;
+	zio_t *pio = zio_walk_parents(zio, &zl);
+	ASSERT(pio);
```

Comment:
I don't see this pio to ever be used.  Is it only to assert that we have some parent?  In such case you could move zio_walk_parents() inside the ASSERT().  The same is in zio_ddt_child_write_ready().

Reply:
Good spot! I've moved the calls inside the asserts.